sdk/core/azure-core-test/src/interceptor_manager.cpp

```diff
@@ -62,7 +63,11 @@ Url InterceptorManager::RedactUrl(Url const& url)
   auto host = url.GetHost();
   auto hostWithNoAccount = std::find(host.begin(), host.end(), '.');
   redactedUrl.SetHost("REDACTED" + std::string(hostWithNoAccount, host.end()));
-  redactedUrl.SetPath(url.GetPath());
+  // replace any uniqueID from the path for a hardcoded id
+  redactedUrl.SetPath(std::regex_replace(
+      url.GetPath(),
+      std::regex("[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}"),
```

Comment:
What is this component: `[89aAbB]`? Do we know that the second last "block" will be one of those 6 characters? And the one before that has to start with `4`? Interesting. Why is that?

Reply:
I got the regex for the universal unique Id. It's defined by some RFC standard already.

Reply:
https://stackoverflow.com/questions/136505/searching-for-uuids-in-text-with-regex

Reply:
Reading through some of this, I don't think we should assume anything about the version of UUID format being used here (unless we know for sure).

For example, the `4` in the third block specifies UUID is formatted as version 4. What guarantees that?

So, I would simplify this regex to the following, which is generic enough to work across all UUID formats:
`[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}`